client/rest-high-level/src/test/java/org/elasticsearch/client/eql/EqlSearchResponseTests.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.eql;
+
+import org.apache.lucene.search.TotalHits;
+import org.elasticsearch.client.AbstractResponseTestCase;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.search.SearchHit;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class EqlSearchResponseTests extends AbstractResponseTestCase<org.elasticsearch.xpack.eql.action.EqlSearchResponse,
+    EqlSearchResponse> {
+
+    static List<SearchHit> randomEvents() {
+        int size = randomIntBetween(1, 10);
+        List<SearchHit> hits = null;
+        if (randomBoolean()) {
+            hits = new ArrayList<>();
+            for (int i = 0; i < size; i++) {
+                hits.add(new SearchHit(i, randomAlphaOfLength(10), new HashMap<>()));
+            }
+        }
+        if (randomBoolean()) {
+            return hits;
+        }
+        return null;
```

Comment:
To me it looks more common to return `hits` and less common to return `null` and I would have switched between them. But that's a personal preference.